### PR TITLE
utils: added common implementation of is_loopback call

### PR DESCRIPTION
**Pull request by hoopoepg**
- implementation of is_loopback_addr call is moved
  to common code section
- added windows implementation
- added IP V6 address processing

Signed-off-by: Oblomov, Sergey <sergey.oblomov@intel.com>

### CODE REVIEW

File: prov/gni/test/cm.c [hunk 1]
```diff
@@ -117,8 +117,7 @@ int cm_local_ip(struct sockaddr_in *sa)
 		/* Return first non loopback interface. */
 		if (ifa->ifa_addr &&
 		    ifa->ifa_addr->sa_family == AF_INET &&
-		    ((struct sockaddr_in *)(ifa->ifa_addr))->sin_addr.s_addr !=
-		     inet_addr("127.0.0.1")) {
+		    ofi_is_loopback_addr(ifa->ifa_addr)) {
```
Comment (shefty) [inline prov/gni/test/cm.c]: @hppritcha @sungeunchoi  - please note change in gni tests

**Comment by shefty** [comment]
The loopback checks don't work on OS X -- see travis.  :(  I don't have access to OS X to help, and can't easily find the in.h header file for it online.

**Comment by hoopoepg** [comment]
I will try to find OS X in office... will fix later

File: prov/gni/test/cm.c (hunk 1 above)
Comment (shefty) [inline prov/gni/test/cm.c]: This check needs to be reversed.  The original check was != loopback.

Comment (hoopoepg) [inline prov/gni/test/cm.c]: yes, my fault...
updated

thank you

File: prov/gni/test/cm.c [hunk 2]
```diff
@@ -117,8 +117,7 @@ int cm_local_ip(struct sockaddr_in *sa)
 		/* Return first non loopback interface. */
 		if (ifa->ifa_addr &&
 		    ifa->ifa_addr->sa_family == AF_INET &&
-		    ((struct sockaddr_in *)(ifa->ifa_addr))->sin_addr.s_addr !=
-		     inet_addr("127.0.0.1")) {
+		    !ofi_is_loopback_addr(ifa->ifa_addr)) {
```
Comment (shefty) [inline prov/gni/test/cm.c]: Thanks - can someone from Cray ack this change please?  @sungeunchoi @hppritcha @jswaro

Comment (jswaro) [inline prov/gni/test/cm.c]: I'll take a look at it.

Comment (jswaro) [inline prov/gni/test/cm.c]: Looks fine to me. None of the CM tests failed.

**Pull request merged**